pkg/skaffold/deploy/util.go

```diff
@@ -83,12 +93,34 @@ func getObjectNamespaceIfDefined(doc []byte, ns string) (string, error) {
 	return ns, nil
 }
 
-func dumpToFileOrWriter(renderedManifests string, filepath string, manifestOut io.Writer) error {
-	if filepath == "" {
+// Outputs rendered manifests to a file, a writer or a GCS bucket.
+func outputRenderedManifests(renderedManifests string, output string, manifestOut io.Writer) error {
+	if output == "" {
```

Comment:
nit: I would probably use a switch here because it's clearer that we have multiple exclusive cases.